compiler/frontend/src/org/jetbrains/kotlin/resolve/calls/inference/ConstraintSystemImpl.kt

```diff
@@ -175,8 +188,9 @@ internal class ConstraintSystemImpl(
 
     override fun toBuilder(filterConstraintPosition: (ConstraintPosition) -> Boolean): ConstraintSystem.Builder {
         val result = ConstraintSystemBuilderImpl()
-        for ((typeParameter, typeBounds) in allTypeParameterBounds) {
-            result.allTypeParameterBounds.put(typeParameter, typeBounds.filter(filterConstraintPosition))
+        for (typeParameter in allTypeParameterBounds.keys) {
+            val typeBounds = allTypeParameterBounds[typeParameter]!!
```

Comment:
This can use `entries`, too:
`for ((typeParameter, typeBounds) in allTypeParameterBounds.entries) {`

Reply:
done